floodviz/static/js/map.js

```diff
@@ -1,116 +1,156 @@
+			return options.proj.forward([lambda, phi].map(radiansToDegrees));
+		};
+		project.invert = function (x, y) {
+			return options.proj.inverse([x, y]).map(degreesToRadians);
+		};
+		//Define map projection
+		var projection = d3.geoProjection(project);
+		// Give projection initial rotation and scale
+		projection.scale(1).translate([0, 0]);
+		//Define path generator
+		var path = d3.geoPath().projection(projection);
+		//Create SVG element
+		var svg = d3.select(options.div_id)
+			.append("svg")
+			.attr("width", options.width)
+			.attr("height", options.height);
+		// Tooltip
+		var maptip = d3.select("body")
+			.append("div")
+			.attr("id", "maptip");
+		
+		/**
+		 * Add circles to the map.
+		 * @param data The geojson to be added to the svg
+		 * @param classname The class to be given to each element for use in CSS
+		 * @param radius The radius of each circle. This cannot be set from CSS
+		 */
+		var add_circles = function (data, classname, radius) {
+			var group = svg.append("g");
+			group.selectAll("circle")
+				.data(data.features)
+				.enter()
+				.append("circle")
+				.attr("r", radius)
+				.attr("transform", function (d) {
+					return "translate(" + projection(d.geometry.coordinates) + ")";
+				})
+				.attr("id", function(d) {
+					if (classname === 'gage-point') {
+						return 'map' + d.properties.id;
+					} else { return '';}
+				})
+				.attr("class", classname);
+			return (group);
+		};
+		/**
+		 * Add paths to the map
+		 * @param data The geojson to be added to the svg
+		 * @param classname The class to be given to each element for use in CSS
+		 */
+		var add_paths = function (data, classname) {
+			var group = svg.append("g");
+			group.selectAll("path")
+				.data(data.features)
+				.enter()
+				.append("path")
+				.attr("d", path)
+				.attr("class", classname);
+		};
+		
+		/**
+		 * Initalize the Map
+		 */
+		self.init = function() {
+			
+			// set bounding box to values provided
+			var b = path.bounds(options.bounds);
+			var s = options.scale / Math.max((b[1][0] - b[0][0]) / options.width, (b[1][1] - b[0][1]) / options.height);
+			var t = [(options.width - s * (b[1][0] + b[0][0])) / 2, (options.height - s * (b[1][1] + b[0][1])) / 2];
+			// Update the projection
+			projection.scale(s).translate(t);
+			// Add layers
+			add_paths(options.bg_data, "background");
+			add_paths(options.rivers_data, "river");
+			add_circles(options.ref_data, "ref-point", 2);
+			// Add sites and bind events for site hovers
+			var sites = add_circles(options.site_data, "gage-point", 3);
+			sites.selectAll("circle")
+				.on('mousemove', function(d) {return self.mousemove(d.properties.name, d.properties.id)})
+				.on("mouseout", function() {return self.mouseout()});
+			// Debug points
+			if (FV.mapinfo.debug) {
+				add_circles(options.bounds, "debug-point", 3)
+			}
+		};
+		/**
+		 * Handle all mouse over movements for calling element. 
+		 */
+		self.mousemove = function (sitename, sitekey) {
```

Comment:
See comments from hydrograph.js. Name of the function should describe how it affects the visualization. This does not set up the interaction itself and has nothing to do with mousemove.